controllers/machinehealthcheck_controller.go

```diff
@@ -237,31 +238,53 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, cluster *c
 			EventRemediationRestricted,
 			"Remediation restricted due to exceeded number of unhealthy machines (total: %v, unhealthy: %v, maxUnhealthy: %v)",
 			totalTargets,
-			totalTargets-currentHealthy,
+			m.Status.CurrentHealthy,
 			m.Spec.MaxUnhealthy,
 		)
+		for _, t := range append(healthy, unhealthy...) {
+			if err := t.patchHelper.Patch(ctx, t.Machine); err != nil {
+				logger.Error(err, "Failed to patch machine status", "machine", t.Machine.Name)
+				return ctrl.Result{}, err
+			}
+		}
 		return reconcile.Result{Requeue: true}, nil
 	}
 	logger.V(3).Info(
 		"Remediations are allowed",
 		"total target", totalTargets,
 		"max unhealthy", m.Spec.MaxUnhealthy,
-		"unhealthy targets", totalTargets-currentHealthy,
+		"unhealthy targets", len(unhealthy),
 	)
 
-	// remediate
+	// mark for remediation
 	errList := []error{}
-	for _, t := range needRemediationTargets {
+	for _, t := range unhealthy {
 		logger.V(3).Info("Target meets unhealthy criteria, triggers remediation", "target", t.string())
-		if err := t.remediate(ctx, logger, r.Client, r.recorder); err != nil {
-			logger.Error(err, "Error remediating target", "target", t.string())
-			errList = append(errList, err)
+
+		conditions.MarkFalse(t.Machine, clusterv1.ConditionOwnerRemediated, "NeverRun", clusterv1.ConditionSeverityWarning, "")
```

Comment:
```suggestion
		conditions.MarkFalse(t.Machine, clusterv1.ConditionOwnerRemediated, "WaitingForRemediation", clusterv1.ConditionSeverityWarning, "")
```
Also, I'm +1 for having condition Reason as a constant in the API

Reply:
Also what about adding in the message something documenting that this condition was triggered by MHC